src/UIKit/UISegmentedControl.cs

```diff
@@ -24,29 +24,71 @@
 
 namespace XamCore.UIKit {
 	public partial class UISegmentedControl {
-		public UISegmentedControl (object [] args) : base (NSObjectFlag.Empty)
+		public UISegmentedControl (params object [] args) : this (FromObjects (args))
+		{
+		}
+
+		static NSArray FromObjects (object[] args)
 		{
 			if (args == null)
-				throw new ArgumentNullException ("args");
+				throw new ArgumentNullException (nameof (args));
 
 			NSObject [] nsargs = new NSObject [args.Length];
 			
 			for (int i = 0; i < args.Length; i++){
 				object a = args [i];
-
 				if (a == null)
-					throw new ArgumentNullException (String.Format ("Element {0} in args is null", i));
-				
-				if (a is string)
-					nsargs [i] = new NSString ((string) a);
-				else if (a is UIImage)
-					nsargs [i] = (UIImage) a;
+					throw new ArgumentNullException ($"Element {i} in args is null");
+
+				var s = (a as string);
+				if (s != null) {
+					nsargs [i] = new NSString (s);
+					continue;
+				}
+
+				var ns = (a as NSString);
+				if (ns != null) {
+					nsargs [i] = ns;
+					continue;
+				}
+
+				var img = (a as UIImage);
+				if (img != null)
+					nsargs [i] = img;
 				else
-					throw new ArgumentException (String.Format ("non-string or UIImage at position {0} with type {1}", i, a.GetType ()));
-			}
-			using (NSArray nsa = NSArray.FromNSObjects (nsargs)){
-				Handle = InitWithItems (nsa.Handle);
+					throw new ArgumentException ("Non-string or UIImage at position {i} with type {a.GetType ()}");
 			}
+			return NSArray.FromNSObjects (nsargs);
+		}
+
+		public UISegmentedControl (params UIImage [] images) : this (FromNSObjects (images))
+		{
+		}
+
+		public UISegmentedControl (params NSString [] strings) : this (FromNSObjects (strings))
+		{
+		}
+
+		static NSArray FromNSObjects (NSObject [] items)
+		{
+			// items will never be null / empty as the callers have `params`
```

Comment:
You can do:

     new UISegmentedControl ((NSString[]) null)

and `items` will be null.